tests/test_workflow_fmriprep.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import json
+from pathlib import Path
+
+import pytest
+from .conftest import global_config_for_testing
+
+import nipoppy.workflow.logger as my_logger
+import nipoppy.workflow.proc_pipe.fmriprep.run_fmriprep as fmriprep_module
+from nipoppy.workflow.proc_pipe.fmriprep.run_fmriprep import run, run_fmriprep
+
+
+def create_dummy_fs_license(pth: Path) -> None:
+    pth = pth / "freesurfer"
+    pth.mkdir(parents=True, exist_ok=True)
+    license_file = pth / "license.txt"
+    license_file.write_text("dummy content")
+
+
+def dummy_bids_filter_file() -> Path:
+    """TODO probably don't want the bids filter file to be in the module directory"""
+    pth = Path(fmriprep_module.__file__).parent
+    return pth / "bids_filter.json"
+
+
+def create_dummy_bids_filter() -> None:
+    with open(dummy_bids_filter_file(), "w") as f:
+        json.dump({"dummy": "dummy"}, f)
+
+
+def delete_dummy_bids_filter() -> None:
+    dummy_bids_filter_file().unlink(missing_ok=True)
+
+
+@pytest.mark.parametrize("use_bids_filter", [True, False])
+@pytest.mark.parametrize("anat_only", [True, False])
+def test_run(tmp_path, use_bids_filter, anat_only):
+    create_dummy_fs_license(tmp_path)
+
+    # TODO because this is set up and torn down after each test
+    # we probably want to turn this into fixture
+    create_dummy_bids_filter()
+
+    participant_id = "01"
+
+    session_id = "01"
+
+    log_file = tmp_path / "fmriprep.log"
+    logger = my_logger.get_logger(log_file)
+
+    run(
+        participant_id=participant_id,
+        global_configs=global_config_for_testing(tmp_path),
+        session_id=session_id,
+        output_dir=tmp_path,
+        use_bids_filter=use_bids_filter,
+        anat_only=anat_only,
+        logger=logger,
+    )
+
+    delete_dummy_bids_filter()
+
+
+@pytest.mark.parametrize("use_bids_filter", [True, False])
+@pytest.mark.parametrize("anat_only", [True, False])
+def test_run_fmriprep(tmp_path, use_bids_filter, anat_only):
+    log_file = tmp_path / "fmriprep.log"
+
+    bids_dir = "bids_dir"
+    fmriprep_dir = tmp_path / "fmriprep_dir"
+    fs_dir = "fs_dir"
+    templateflow_dir = "templateflow_dir"
+    participant_id = "01"
```

Comment:
Do expect to have bids formatted id?

Reply:
well I do not think there are any checks on that at the moment, no?

Reply:
No because fmriprep doesn't require the `sub-` prefix. But I think we should standardize all "run script" calls to comply with the bids subject id naming.

Reply:
probably a separate PR to make sure it is participant_id is alphanumeric